staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -129,6 +131,14 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 		audit.LogRequestObject(ae, obj, scope.Resource, scope.Subresource, scope.Serializer)
 
 		userInfo, _ := request.UserFrom(ctx)
+
+		// On create, get name from new object if unset
+		if len(name) == 0 {
+			if _, name, err = scope.Namer.ObjectName(obj); err != nil {
```

Comment:
Not all submitted API objects contain names (for example, non-persisted review objects and objects using `GenerateName`)

We've already enforced `includeName` above, so ignore errors here.

Reply:
also, to handle the GeneratedName case, in AdmissionToValidateObjectFunc, if `staticAttributes.GetName()` is empty, shouldn't we attempt to override with obj.GetName()? A testcase with generate name would be good (mutating admission wouldn't currently get the name attribute, validating admission would)